docs/modules/architecture/pages/compact-binary-specification.adoc

```diff
@@ -0,0 +1,719 @@
+|
+|An int32 that can also be null
+|no
+|nullable-int64
+|Long
+|optional<int64_t>
+|long?
+|typing.Optional[int]
+|Long \| null
+|*int64
+|
+|An int64 that can also be null
+|no
+|nullable-float32
+|Float
+|optional<float>
+|float?
+|typing.Optional[float]
+|number \| null
+|*float32
+|
+|A float32 that can also be null
+|no
+|nullable-float64
+|Double
+|optional<double>
+|double?
+|typing.Optional[float]
+|number \| null
+|*float64
+|
+|A double that can also be null
+|no
+|nullable-boolean[]
+|Boolean[] 
+|optional<std::vector<optional<bool>>> 
+|bool?[]?
+|typing.Optional[typing.List[typing.Optional[bool]]]
+|(boolean \| null)[] \| null
+|[]*bool
+|
+|Array of nullable booleans
+|no
+|nullable-int8[]
+|Byte[] 
+|optional<std::vector<optional<int8_t>>> 
+|sbyte?[]?
+|typing.Optional[typing.List[typing.Optional[int]]]
+|(number \| null)[] \| null
+|[]*int8
+|
+|Array of nullable int8s
+|no
+|nullable-int16[]
+|Short[] 
+|optional<std::vector<optional<int16_t>>>
+|short?[]?
+|typing.Optional[typing.List[typing.Optional[int]]]
+|(number \| null)[] \| null
+|[]*int16
+|
+|Array of nullable i1int6s
+|no
+|nullable-int32[]
+|Integer[] 
+|optional<std::vector<optional<int32_t>>>
+|int?[]?
+|typing.Optional[typing.List[typing.Optional[int]]]
+|(number \| null)[] \| null
+|[]*int32
+|
+|Array of nullable int32s
+|no
+|nullable-int64[]
+|Long[] 
+|optional<std::vector<optional<int64_t>>> 
+|long?[]?
+|typing.Optional[typing.List[typing.Optional[int]]]
+|(Long \| null)[] \| null
+|[]*int64
+|
+|Array of nullable int64s
+|no
+|nullable-float32[]
+|Float[] 
+|optional<std::vector<optional<float>>> 
+|float?[]?
+|typing.Optional[typing.List[typing.Optional[float]]]
+|(number \| null)[] \| null
+|[]*float32
+|
+|Array of nullable float32s
+|no
+|nullable-float64[]
+|Double[] 
+|optional<std::vector<optional<double>>> 
+|double?[]?
+|typing.Optional[typing.List[typing.Optional[float]]]
+|(number \| null)[] \| null
+|[]*float64
+|
+|Array of nullable float64
+|no
+|===
+
+
+=== Type Ids
+
+Each type supported in the wire format has its type id. The type ids are used while constructing the schemas, performing type checks for user access to fields, and are exposed to the user as a public API.
+
+To distinguish different types supported in the wire-level format, a new enum called `FieldKind` is introduced which has different ids for different types compared to the `FieldType` enum.
+
+The `FieldType` enum has different ways to represent ids for types and their arrays and has some unfortunate public methods which would not fit the new format. (for example `FieldType#DATE` returns `getTypeSize` wrong for the new format). It is desired to have a new enum that does not have those problems and is more open to type additions in the future.
+
+Note that, the new `FieldKind` enum with those new ids will be used in the `GenericRecord` API for both `Portable` and the new format. `FieldType` and `Portable` will be deprecated in the future.
+
+=== Unsigned Integers
```

Comment:
We don't have support for unsigned integers right now

Reply:
removing the section.